nts-pool-ke/src/lib.rs

```diff
@@ -265,29 +260,39 @@ async fn pick_nts_ke_server<'a>(
     }
 }
 
+// which errors reported by a server are "fatal" for us, and which just mean that
+// the server should be ignored?
+fn is_fatal_error(error: &KeyExchangeError) -> bool {
+    matches!(error, KeyExchangeError::BadRequest)
+}
+
 async fn pick_nts_ke_servers<'a>(
     connector: &TlsConnector,
     servers: &'a [config::KeyExchangeServer],
     selected_algorithm: AeadAlgorithm,
     denied_servers: &[String],
 ) -> Result<(&'a str, u16, ServerName), KeyExchangeError> {
+    let mut downstream_error = false;
+
     for server in servers {
         if denied_servers.contains(&server.domain) {
             continue;
         }
 
         match pick_nts_ke_server(connector, server, selected_algorithm).await {
             Ok(x) => return Ok(x),
-            Err(e) => match e {
-                KeyExchangeError::Io(e) if e.kind() == ErrorKind::ConnectionRefused => continue,
-                _ => return Err(e),
-            },
+            Err(e) if is_fatal_error(&e) => return Err(e),
```

Comment:
**Note for the reviewer**

I am quite unsure if having `is_fatal_error` is a wise idea at all. It was clear that the original code was too strict (already prevented a normal test setup from operating in case one NTSKE server presented the wrong certificate, for example).

But imagine if a NTS-POOL consists of many servers; with this PR, if one of them sends a `BadRequest` it still can bring the whole pool down, which is a DoS risk. But I find it hard to identify legitimate use cases where we would want to stop trying alternate NTSKE servers.